ydb/src/client_topic/topicwriter/writer.rs

```diff
@@ -0,0 +1,405 @@
+// Experimental
+//
+// Notice: This API is EXPERIMENTAL and may be changed or removed in a later release.
+
+use crate::client_topic::common::grpc_stream_wrapper::AsyncGrpcStreamWrapper;
+use crate::client_topic::topicwriter::init_writer::RawInitResponse;
+use crate::client_topic::topicwriter::message::TopicWriterMessage;
+use crate::client_topic::topicwriter::message_write_status::{MessageWriteStatus, WriteAck};
+use crate::client_topic::topicwriter::writer_options::TopicWriterOptions;
+use crate::client_topic::topicwriter::writer_reception_queue::{
+    TopicWriterReceptionQueue, TopicWriterReceptionTicket, TopicWriterReceptionType,
+};
+use crate::grpc_connection_manager::GrpcConnectionManager;
+use crate::grpc_wrapper::raw_errors::{RawError, RawResult};
+use crate::grpc_wrapper::raw_topic_service::common::codecs::RawSupportedCodecs;
+use crate::{grpc_wrapper, YdbError, YdbResult};
+use futures_util::{StreamExt, TryStreamExt};
+use std::borrow::{Borrow, BorrowMut};
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::future::Future;
+use std::ops::Deref;
+use std::pin::Pin;
+use std::sync::{Arc, Mutex, MutexGuard};
+use std::task::{Context, Poll};
+use std::time::Instant;
+use std::time::{Duration, UNIX_EPOCH};
+use tokio::select;
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::Receiver;
+use tokio::task::JoinHandle;
+use tokio::time::error::Elapsed;
+use tokio::time::timeout;
+use tokio_util::sync::CancellationToken;
+use ydb_grpc::ydb_proto::topic::stream_write_message;
+use ydb_grpc::ydb_proto::topic::stream_write_message::from_client::ClientMessage;
+use ydb_grpc::ydb_proto::topic::stream_write_message::from_server::ServerMessage;
+use ydb_grpc::ydb_proto::topic::stream_write_message::init_request::Partitioning;
+use ydb_grpc::ydb_proto::topic::stream_write_message::write_request::{message_data, MessageData};
+use ydb_grpc::ydb_proto::topic::stream_write_message::{
+    FromClient, FromServer, InitRequest, WriteRequest,
+};
+
+pub enum TopicWriterState {
```

Comment:
In the crate "state" used for more complex structs, included full state of object.

For topic writer state is: cancelation token, writer/receiver loops, this enum, confirmation queue, connection manager...

all, what must be copy in clone func.

I suggest to rename the type and field, for example for "mode" or "stop reason" with: None, OK, Error variants.